### DISCUSSION TITLE
can not get or change default response header Server

### DISCUSSION THREAD

**Discussion by badiku** [RealTimeLogic/BAS#7] [Q&A] [answered]
hi, just tried mako server.

I got blank string for response header Server, 
while I cound see from Chrome devtools that it's value is MakoServer.net
```
<?lsp 
response:write('Server: ',response:containsheader('Server'))  -- got blank string
 ?>
```
 

and if I change header Server to other strings, 
I cound get this string,
but I could see from Chrome devtools that it's value is MakoServer.net and this string.
there're two values for header Server now. why?
```
<?lsp 
response:setheader('Server', 'myServer')
response:write('Server: ',response:containsheader('Server'))  -- got myServer, but MakoServer.net still exists
 ?>
```

![d](https://user-images.githubusercontent.com/30396109/193549075-d959bb2c-0d4b-47d1-a387-e70df5dc03b3.jpg)

  

is this a bug , or it is a feature  ?

doc says that `the new value overwrites the previous one` :

> response:setheader(name [, value])
>     Sets a HTTP response header with the given name and value. If the header had already been set, the new value overwrites the previous one. The response:containsheader method can be used to test for the presence of a header before setting its value. If the value is not provided then the named header is removed from the response.
>     setheader() should be called before the response has been committed to the client, (before response body output has been flushed). If the response already has been committed, this method returns false.

**Comment by surfskidude** [comment, marked as answer]
This is a feature. Some headers are reserved. The following works:
response:setheader('Hello', 'World')
trace(response:containsheader'Hello')
Changing the server name requires the full Barracuda App Server SDK and is an option for commercial entities purchasing the full source code.
